templates/_helpers.tpl

```diff
@@ -98,6 +98,17 @@ Get the mastodon secret.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Get the smtp secret.
+*/}}
+{{- define "mastodon.smtp.secretName" -}}
+{{- if .Values.mastodon.smtp.existingSecret }}
+    {{- printf "%s" (tpl .Values.mastodon.smtp.existingSecret $) -}}
+{{- else -}}
+    {{- printf "%s-smtp" (include "common.names.fullname" .) -}}
```

Comment:
I used `common.names.fullname` following the pattern of `mastodon.secretName`, but I don't think this is something we should do. The `common.names.fullname` helper is actually defined in the redis/postgres charts (a bitnami standard, I believe), so we may get run into problems if the definition of this template changes, or the subcharts are swapped out in the future.

Reply:
I agree, this is not a good idea to use this helper.

If you want to open a PR moving away from it, I will definitely have a look!